src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -42,27 +46,47 @@
 
 /**
  * Scanner that will scan down all forks when we hit parallel blocks before continuing, but generally runs in linear order
- * <p/>Think of it as the opposite of {@link DepthFirstScanner}.
+ * <p></p>Think of it as the opposite of {@link DepthFirstScanner}.
  *
- * <p/>This is a fairly efficient way to visit all FlowNodes, and provides three useful guarantees:
+ * <p></p>This is a fairly efficient way to visit all FlowNodes, and provides three useful guarantees:
  * <ul>
  *   <li>Every FlowNode is visited, and visited EXACTLY ONCE (not true for LinearScanner)</li>
  *   <li>All parallel branches are visited before we move past the parallel block (not true for DepthFirstScanner)</li>
  *   <li>For EVERY block, the BlockEndNode is visited before the BlockStartNode (not true for DepthFirstScanner, with parallels)</li>
  * </ul>
  *
- * <p/>The big advantages of this approach:
+ * <p></p>The big advantages of this approach:
  * <ul>
  *     <li>Blocks are visited in the order they end (no backtracking) - helps with working a block at a time</li>
  *     <li>Points are visited in linear order within a block (easy to use for analysis)</li>
  *     <li>Minimal state information needed</li>
  *     <li>Branch information is available for use here</li>
  * </ul>
  *
- * @author <samvanoort@gmail.com>Sam Van Oort</samvanoort@gmail.com>
+ * @author Sam Van Oort
  */
+@NotThreadSafe
 public class ForkScanner extends AbstractFlowScanner {
 
+    @CheckForNull
+    public NodeType getCurrentType() {
+        return currentType;
+    }
+
+    @CheckForNull
+    public NodeType getNextType() {
+        return nextType;
+    }
+
+    /** Used to recognize special nodes */
+    public enum NodeType {
+        NORMAL,
+        PARALLEL_START,
+        PARALLEL_END,
+        PARALLEL_BRANCH_START,
```

Comment:
I _think_ I know what you are referring to here, but it would be best to add Javadoc for everyone else.